### PR TITLE
Add mapping slot API to cheatcodes (rebased)

### PR DESCRIPTION
**Pull request by mattsse**
See https://github.com/foundry-rs/foundry/pull/4710

### CODE REVIEW

File: testdata/cheats/Cheats.sol [hunk 1]
```diff
@@ -578,4 +578,14 @@ interface Cheats {
...
+
+    function startMappingRecording() external;
+
+    function stopMappingRecording() external;
+
+    function getMappingLength(address target, bytes32 slot) external returns (uint256);
+
+    function getMappingSlotAt(address target, bytes32 slot, uint256 idx) external returns (bytes32);
+
+    function getMappingKeyAndParentOf(address target, bytes32 slot) external returns (bool, bytes32, bytes32);
```
Comment (Evalir) [inline testdata/cheats/Cheats.sol:590]: can we leave some docs here?

**Review by Evalir — APPROVED** [review]
gg, just a comment on docs!

thank you @clouds56, as customary please follow up with a book & forge-std pr!

**Comment by mattsse** [comment]
@clouds56 it'd be great to get some docs here as mention in #4710 that's the only thing that's missing, I really would like to get this in

**Review by Evalir — APPROVED** [review]
never got a proper review for this until now btw, but pretty clean ^^

**Comment by mds1** [comment]
Awesome that this was merged, @Evalir do you mind opening a PR to add this to forge-std and update https://github.com/foundry-rs/forge-std/pull/423 to used named params as well?

**Comment by Evalir** [comment]
yep sir on it!

**Pull request merged**